File: r/bob/bob.R
bob <- function(input) {
  input <- trimws(input)
  if (input == "") return("Fine. Be that way!")

  asking   <- endsWith(input, "?")
  shouting <- grepl("[[:alpha:]]", input) &&
            ! grepl("[[:lower:]]", input)

  if (shouting && asking) "Calm down, I know what I'm doing!"
  else if (shouting)      "Whoa, chill out!"
  else if (asking)        "Sure."
  else                    "Whatever."
}

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        return with(input.trimEnd()) {
            val silence  = this.isEmpty()
            val asking   = this.endsWith('?')
            val shouting = this.contains(Regex("\\p{Upper}"))
                       && !this.contains(Regex("\\p{Lower}"))
            when {
                silence -> "Fine. Be that way!"
                shouting && asking -> "Calm down, I know what I'm doing!"
                shouting -> "Whoa, chill out!"
                asking -> "Sure."
                else -> "Whatever."
            }
        }
    }
}
